### PR TITLE
Support protocol handling for "Start Work" from the chrome extension

### PR DESCRIPTION
**Pull request by ppezaris**
**This PR Addresses:**  
[chrome & firefox PR extension](https://trello.com/c/Mx5sA7uZ/4649-chrome-firefox-pr-extension)

### CODE REVIEW

**Comment by ppezaris** [comment]
i changed the generic provider connection to make it possible to use it w/trello as well -- not sure if this is a good idea or not.

**Comment by ppezaris** [comment]
[Trello powerup to "start work"](https://trello.com/c/3aiOcTJY/4276-trello-powerup-to-start-work)

File: shared/agent/src/providers/github.ts [hunk 1]
```diff
@@ -970,6 +970,22 @@ export class GitHubProvider extends ThirdPartyIssueProviderBase<CSGitHubProvider
 		return response;
 	}
 
+	async setAssigneeOnIssue(request: { issueId: string; assigneeId: string; onOff: boolean }) {
```
Comment (bcanzanella) [inline shared/agent/src/providers/github.ts:973]: test

File: shared/ui/index.tsx [hunk 2]
```diff
@@ -400,6 +401,58 @@ function listenForEvents(store) {
...
+							HostApi.instance
+								.send(ExecuteThirdPartyRequestUntypedType, {
+									method: "selfAssignCard",
+									providerId: card.providerId,
+									params: { cardId: card.id }
+								})
+								.then(() => {
+									store.dispatch(setCurrentReview(""));
+									store.dispatch(setCurrentPullRequest(""));
+									store.dispatch(setStartWorkCard(card));
```
Comment (bcanzanella) [inline shared/ui/index.tsx:419]: i added a `clearCurrentPullRequest` for this

File: shared/ui/index.tsx [hunk 3]
```diff
@@ -400,6 +401,58 @@ function listenForEvents(store) {
...
+										HostApi.instance
+											.send(ExecuteThirdPartyRequestUntypedType, {
+												method: "setAssigneeOnIssue",
+												providerId: route.query.providerId,
+												params: { issueId: issue.id, assigneeId: issue.viewer.id, onOff: true }
+											})
+											.then(() => {
+												store.dispatch(setCurrentReview(""));
+												store.dispatch(setCurrentPullRequest(""));
+												store.dispatch(setStartWorkCard(issue));
```
Comment (bcanzanella) [inline shared/ui/index.tsx:440]: i added a `clearCurrentPullRequest` for this

**Review by bcanzanella — CHANGES_REQUESTED** [review]
looks good. this will need to be rebased from dev before merging. i'll check it out

File: shared/ui/index.tsx (hunk 3 above)
Comment (bcanzanella) [reply shared/ui/index.tsx:453]: i updated it

File: shared/ui/index.tsx (hunk 2 above)
Comment (bcanzanella) [reply shared/ui/index.tsx:432]: i updated it

**Pull request merged**